jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java

```diff
@@ -179,6 +181,18 @@ public void testAsChildOfAcceptNull() {
     assertTrue(jaegerSpan.getReferences().isEmpty());
   }
 
+  @Test
+  public void testAsChildOfIgnoreUnexpectedContextImpl() {
+    tracer = new JaegerTracer.Builder("foo")
+        .withReporter(new InMemoryReporter())
+        .withSampler(new ConstSampler(true))
+        .build();
+
+    JaegerSpan jaegerSpan = tracer.buildSpan("foo").asChildOf(NoopSpan.INSTANCE.context()).start();
```

Comment:
I think it more belongs to span test class

Reply:
done

Reply:
actually in tracer test class there is `testAsChildOfAcceptNull` but I think it more belongs here even span builder is in tracer class.

Reply:
sorry unclear - do you mean move the test back to the `JaegerTracerTest` class?

Reply:
No it's fine here, there is different test in tracer test class which is similar to this tests. I think that one also belongs here